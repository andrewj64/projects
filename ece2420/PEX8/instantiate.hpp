bTree<std::string, bTree<std::string, int> > outer;
bTree<std::string, int> inner;
std::shared_ptr<KeyVal<std::string, int> > pKeyVal(new bTree<std::string, int>); 
bTree<int, std::string> numeric;
