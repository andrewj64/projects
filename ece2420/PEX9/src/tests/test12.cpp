#include "numbers.hpp"
#include <iostream>
#include <vector>
#include "include.hpp"

void bll() {
#include "instantiate.hpp"
#include "rbti.hpp"
	std::cout << "+----------------------------------------------------------------------+" << std::endl;
	std::cout << "|                               Basic LL                               |" << std::endl;
	std::cout << "+----------------------------------------------------------------------+" << std::endl;
	std::vector<int> nums = {40, 20, 10};
	insertSet(nums);
}

void blr() {
#include "instantiate.hpp"
#include "rbti.hpp"
	std::cout << "+----------------------------------------------------------------------+" << std::endl;
	std::cout << "|                               Basic LR                               |" << std::endl;
	std::cout << "+----------------------------------------------------------------------+" << std::endl;
	std::vector<int> nums = {40, 20, 30};
	insertSet(nums);
}

void brl() {
#include "instantiate.hpp"
#include "rbti.hpp"
	std::cout << "+----------------------------------------------------------------------+" << std::endl;
	std::cout << "|                               Basic RL                               |" << std::endl;
	std::cout << "+----------------------------------------------------------------------+" << std::endl;
	std::vector<int> nums = {40, 60, 50};
	insertSet(nums);
}

void brr() {
#include "instantiate.hpp"
#include "rbti.hpp"
	std::cout << "+----------------------------------------------------------------------+" << std::endl;
	std::cout << "|                               Basic RR                               |" << std::endl;
	std::cout << "+----------------------------------------------------------------------+" << std::endl;
	std::vector<int> nums = {40, 60, 70};
	insertSet(nums);

}

int main() {
	bll();
	blr();
	brl();
	brr();
}
