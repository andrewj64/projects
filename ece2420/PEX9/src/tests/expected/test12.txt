+----------------------------------------------------------------------+
|                               Basic LL                               |
+----------------------------------------------------------------------+
Insert 40
{K:40, C:BLACK, L:0, PK:NULL}

Insert 20
{K:40, C:BLACK, L:0, PK:NULL}
{K:20, C:RED, L:1, PK:40}

Insert 10
{K:20, C:BLACK, L:0, PK:NULL}
{K:10, C:RED, L:1, PK:20}{K:40, C:RED, L:1, PK:20}

+----------------------------------------------------------------------+
|                               Basic LR                               |
+----------------------------------------------------------------------+
Insert 40
{K:40, C:BLACK, L:0, PK:NULL}

Insert 20
{K:40, C:BLACK, L:0, PK:NULL}
{K:20, C:RED, L:1, PK:40}

Insert 30
{K:30, C:BLACK, L:0, PK:NULL}
{K:20, C:RED, L:1, PK:30}{K:40, C:RED, L:1, PK:30}

+----------------------------------------------------------------------+
|                               Basic RL                               |
+----------------------------------------------------------------------+
Insert 40
{K:40, C:BLACK, L:0, PK:NULL}

Insert 60
{K:40, C:BLACK, L:0, PK:NULL}
{K:60, C:RED, L:1, PK:40}

Insert 50
{K:50, C:BLACK, L:0, PK:NULL}
{K:40, C:RED, L:1, PK:50}{K:60, C:RED, L:1, PK:50}

+----------------------------------------------------------------------+
|                               Basic RR                               |
+----------------------------------------------------------------------+
Insert 40
{K:40, C:BLACK, L:0, PK:NULL}

Insert 60
{K:40, C:BLACK, L:0, PK:NULL}
{K:60, C:RED, L:1, PK:40}

Insert 70
{K:60, C:BLACK, L:0, PK:NULL}
{K:40, C:RED, L:1, PK:60}{K:70, C:RED, L:1, PK:60}

